### PR TITLE
[KOGITO-1594] - Specify JVM memory based on pod limit

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>
See: https://issues.redhat.com/browse/KOGITO-1594

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: modules/kogito-builder/added/kogito-app-launch.sh [hunk 1]
```diff
@@ -23,23 +23,27 @@ source "${KOGITO_HOME}"/launch/configure.sh
...
+JAVA_OPTS="$(${JBOSS_CONTAINER_JAVA_JVM_MODULE}/java-default-options) $(${JBOSS_CONTAINER_JAVA_JVM_MODULE}/debug-options)"
+
 #############################################
 
 case ${runtime_type} in 
     "quarkus")  
         if [ "${NATIVE^^}" == "TRUE" ]; then
             # shellcheck disable=SC2086
-            exec "${KOGITO_HOME}"/bin/*-runner ${JAVA_OPTIONS} ${KOGITO_QUARKUS_S2I_PROPS} \
+            exec "${KOGITO_HOME}"/bin/*-runner ${JAVA_OPTS} ${JAVA_OPTIONS} ${KOGITO_QUARKUS_S2I_PROPS} \
```
Comment (radtriste) [inline modules/kogito-builder/added/kogito-app-launch.sh]: sorry for nitpicking but `JAVA_OPTS` and `JAVA_OPTIONS` are a bit confusing.
couldn't we set the new env var as `CONTAINER_JAVA_OPTS` ? or `DYNAMIC_RESOURCES_OPTS` instead ?

**Comment by radtriste** [comment]
Jenkins retest this

File: modules/kogito-builder/added/kogito-app-launch.sh (hunk 1 above)
Comment (spolti) [reply modules/kogito-builder/added/kogito-app-launch.sh]: sure thing, let's use DYNAMIC_RESOURCES_OPTS

File: tests/features/common.feature [hunk 2]
```diff
@@ -25,3 +25,44 @@ Feature: Common tests for Kogito images
...
+      | env_json  | {"SCRIPT_DEBUG":"true", "JAVA_MAX_MEM_RATIO": 80, "JAVA_INITIAL_MEM_RATIO": 25} |
+    Then container log should match regex -Xms205m
+    And container log should match regex -Xmx819m
+
+  Scenario: Verify if the DEFAULT MEM RATIO properties are overridden with different values
+    When container is started with args
+      | arg       | value                                                                           |
+      | mem_limit | 1073741824                                                                      |
+      | env_json  | {"SCRIPT_DEBUG":"true", "JAVA_MAX_MEM_RATIO": 50, "JAVA_INITIAL_MEM_RATIO": 10} |
+    Then container log should match regex -Xms51m
```
Comment (sutaakar) [inline tests/features/common.feature]: Shouldn't be -Xms around 102?

Comment (spolti) [reply tests/features/common.feature]: no, it will be 10% of Xmx

File: modules/kogito-builder/added/kogito-app-launch.sh (hunk 1 above)
Comment (spolti) [reply modules/kogito-builder/added/kogito-app-launch.sh]: rename

File: tests/features/common.feature (hunk 2 above)
Comment (sutaakar) [reply tests/features/common.feature]: oh, ok

**Review by sutaakar — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Review by vaibhavjainwiz — APPROVED** [review]

**Pull request merged**